### PR TITLE
Bpmn explorer

### PR DESCRIPTION
**Pull request by NatashaWso2**
BPS-794

### CODE REVIEW

File: components/bpmn/bpmn-explorer-web/src/web/js/actions.js [hunk 1]
```diff
@@ -310,4 +310,41 @@ function processSearch(){
...
+
+            function drawChart(data) {
+                var dataArr=[['Task Key', 'Avg Duration']];
+                for(var i=0;i<data.length;i++){
+                    dataArr.push([data[i][0] , data[i][1]]);
+
+                }
+
+                var data = google.visualization.arrayToDataTable(dataArr);
+
```
Comment (jranabahu) [inline components/bpmn/bpmn-explorer-web/src/web/js/actions.js]: Dont keep unnecessary new lines.

File: components/bpmn/bpmn-explorer-web/src/web/controller/statsController.jag [hunk 2]
```diff
@@ -12,22 +12,30 @@
  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  ~ See the License for the specific language governing permissions and
  ~ limitations under the License.
--->
+-->*/%>
 <%
 	var authSuccess = session.get("authSuccess");
-	if(!authSuccess){
+
+		if(!authSuccess){
```
Comment (cnapagoda) [inline components/bpmn/bpmn-explorer-web/src/web/controller/statsController.jag]: Formatting issue

File: components/bpmn/bpmn-explorer-web/src/web/controller/statsController.jag [hunk 3]
```diff
@@ -12,22 +12,30 @@
...
 		var queryParameters = request.getQueryString();
-		if(queryParameters == "null" || queryParameters == null){
-			response.sendRedirect("login?requestUrl=" + request.getRequestURI());
-		} else{
-			response.sendRedirect("login?requestUrl=" + request.getRequestURI() + "%3F" + request.getQueryString());
-		}
+        	if(queryParameters == "null" || queryParameters == null){
```
Comment (cnapagoda) [inline components/bpmn/bpmn-explorer-web/src/web/controller/statsController.jag]: Code formatting issue

File: components/bpmn/bpmn-explorer-web/src/web/model/statsModel.jag [hunk 4]
```diff
@@ -40,8 +40,149 @@ for(var i=0; i<processNames.data.length; i++){
...
+      for(var i in timeduration.data.data){
+        var timeForOne = timeduration.data.data[i].durationInMillis;
+        var timeInMinutes = ((timeForOne /(1000*60))%60);
+        totalTime = totalTime + timeInMinutes;
+      }
+
+   var averageTime = totalTime/count;
+   var avgRoundedTime = Math.round(averageTime*1000)/1000;
+
+    if(checkForDuplicates(processBarChartAvgTime,processName) == 1){
```
Comment (cnapagoda) [inline components/bpmn/bpmn-explorer-web/src/web/model/statsModel.jag]: Code formatting issue

File: components/bpmn/bpmn-explorer-web/src/web/js/actions.js (hunk 1 above)
Comment (NatashaWso2) [inline components/bpmn/bpmn-explorer-web/src/web/js/actions.js]: Fixed with commit 130406a1db83af11af57067f0f18ad3a87f987ce

File: components/bpmn/bpmn-explorer-web/src/web/controller/statsController.jag (hunk 2 above)
Comment (NatashaWso2) [inline components/bpmn/bpmn-explorer-web/src/web/controller/statsController.jag]: Fixed with commit 130406a1db83af11af57067f0f18ad3a87f987ce

File: components/bpmn/bpmn-explorer-web/src/web/model/statsModel.jag (hunk 4 above)
Comment (NatashaWso2) [inline components/bpmn/bpmn-explorer-web/src/web/model/statsModel.jag]: Fixed with commit 130406a

File: components/bpmn/bpmn-explorer-web/src/web/controller/statsController.jag (hunk 3 above)
Comment (NatashaWso2) [inline components/bpmn/bpmn-explorer-web/src/web/controller/statsController.jag]: This code snippet has been moved to reportController.jag and the formatting issue is fixed there.
Fixed with commit 130406a

**Pull request merged**